File: seckill/src/main/java/cn/edu/jxnu/seckill/access/UserContext.java
package cn.edu.jxnu.seckill.access;

import cn.edu.jxnu.seckill.domain.SeckillUser;

public class UserContext {
    private static ThreadLocal<SeckillUser> userHolder = new ThreadLocal<>();

    public static void setUser(SeckillUser user) {
        userHolder.set(user);
    }

    public static SeckillUser getUser() {
        return userHolder.get();
    }
}

File: src/main/scala/cn/edu/jxnu/seckill/access/UserContext.scala
package cn.edu.jxnu.seckill.access

import cn.edu.jxnu.seckill.domain.SeckillUser

/**
 * 用户上下文
 *
 * @author 梦境迷离.
 * @time 2018年5月19日
 * @version v1.0
 */
package object UserContext {

    private var userHolder: ThreadLocal[SeckillUser] = new ThreadLocal[SeckillUser]()

    def setUser(user: SeckillUser) {
        userHolder.set(user)
    }

    def getUser(): SeckillUser = userHolder.get()

}